src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact{
+    private String value;
+    private boolean isPrivate;
+    public Contact(String contact, boolean isPrivate, String validation, String constrains){
+ 		this.isPrivate = isPrivate;
+        contact = contact.trim();
+        if (!isValid(contact, validation)) {
+            throw new IllegalValueException(constrains);
+        }
+        this.value = contact;
+ 	  }
+ 	  
+ 	  private boolean isValid(String contact, String validation){
+ 	      return test.mathes(contact, validation);
```

Comment:
I think you have blatantly copied and pasted this function without much thought. Note that in the original function, test is a string, and test.matches will match the current string with the validation. Please try to build your program and check that it is error-free before putting up the PR.